pkg/ottl/ottlfuncs/func_contains.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ContainsArguments[K any] struct {
+	Target        ottl.StringGetter[K]
+	Patterns      []string
+	CaseSensitive bool
+}
+
+func NewContainsFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory(
+		"Contains",
+		&ContainsArguments[K]{},
+		createContainsFunction[K],
+	)
+}
+
+func createContainsFunction[K any](
+	_ ottl.FunctionContext,
+	oArgs ottl.Arguments,
+) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ContainsArguments[K])
+
+	if !ok {
+		return nil, errors.New(
+			"NewContainsFactory args must be of type *ContainsArguments[K]",
```

Comment:
Adapt the style of other ottl functions:
```suggestion
			"ContainsFactory args must be of type *ContainsArguments[K]",
```